### PR TITLE
Remove tests for json constructors, remove helper class use FluentAss…

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/test/iothub/device/AzureSecurityCenterForIoTSecurityMessageE2ETests.cs [hunk 1]
```diff
@@ -90,17 +92,20 @@ private TelemetryMessage ComposeD2CSecurityTestMessage()
             return message;
         }
 
-        private static JObject ComposeAzureSecurityCenterForIoTSecurityMessagePayload(string eventId)
+        private static object ComposeAzureSecurityCenterForIoTSecurityMessagePayload(string eventId)
```
Comment (drwill-ms) [inline e2e/test/iothub/device/AzureSecurityCenterForIoTSecurityMessageE2ETests.cs:95]: Updated this to create a .NET object instead, which is serialized on line 84.

File: e2e/test/provisioning/ProvisioningE2ETests.cs [hunk 2]
```diff
@@ -1030,8 +1030,8 @@ public static ProvisioningClientOptions CreateProvisioningClientOptionsFromName(
                                     iothubs,
                                     capabilities)
                                 .ConfigureAwait(false);
-                            Assert.IsTrue(symmetricKeyEnrollmentGroup.Attestation is SymmetricKeyAttestation);
-                            var symmetricKeyAttestation = (SymmetricKeyAttestation)symmetricKeyEnrollmentGroup.Attestation;
+                            symmetricKeyEnrollmentGroup.Attestation.Type.Should().Be(AttestationMechanismType.SymmetricKey);
```
Comment (drwill-ms) [inline e2e/test/provisioning/ProvisioningE2ETests.cs:1033]: I suspect all this stuff got broke when we updated the json constructor stuff. Do these changes look right to you?

File: provisioning/service/tests/Config/AttestationMechanismTests.cs [hunk 3]
```diff
@@ -1,127 +0,0 @@
...
-using System;
-using System.Text;
-using System.Text.Json;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
-
-namespace Microsoft.Azure.Devices.Provisioning.Service.Test
-{
-    [TestClass]
-    [TestCategory("Unit")]
-    public class AttestationMechanismTests
```
Comment (drwill-ms) [inline provisioning/service/tests/Config/AttestationMechanismTests.cs:13]: Most of these tests were validating the json constructor logic... which we no longer have... so, I deleted them.

File: provisioning/service/tests/Config/X509CertificateWithInfoTests.cs [hunk 4]
```diff
@@ -95,75 +62,54 @@ public void X509CertificateWithInfoConstructorThrowsOnNullX509Certificate()
...
-            var error2 = act2.Should().Throw<ProvisioningServiceException>();
-            error2.And.StatusCode.Should().Be(HttpStatusCode.BadRequest);
-            error2.And.IsTransient.Should().BeFalse();
-
-            Action act3 = () => new X509CertificateWithInfo(certificateStringInvalid);
-            var error3 = act3.Should().Throw<ProvisioningServiceException>();
-            error3.And.StatusCode.Should().Be(HttpStatusCode.BadRequest);
-            error3.And.IsTransient.Should().BeFalse();
-#pragma warning restore CA1806 // Do not ignore method results
+            Action act = () => _ = new X509CertificateWithInfo(certificateNull);
```
Comment (drwill-ms) [inline provisioning/service/tests/Config/X509CertificateWithInfoTests.cs:65]: I updated these tests to use FluentAssertions, so I could delete the TestAssert.cs file.

File: e2e/test/iothub/device/MethodE2ETests.cs [hunk 5]
```diff
@@ -344,7 +344,7 @@ public async Task Method_OpenCloseOpenDeviceReceivesDirectMethods_AmqpTcp()
 
             VerboseTestLogger.WriteLine($"{nameof(ServiceSendMethodAndVerifyResponseAsync)}: Method status: {response.Status}.");
             response.Status.Should().Be(200);
-            JsonSerializer.Serialize(response.Payload).Should().Be(JsonSerializer.Serialize(respJson));
+            JsonSerializer.Serialize(response.PayloadAsString).Should().Be(JsonSerializer.Serialize(respJson));
```
Comment (abhipsaMisra) [inline e2e/test/iothub/device/MethodE2ETests.cs:347]:
```suggestion
            response.PayloadAsString.Should().Be(JsonSerializer.Serialize(respJson));
```

File: e2e/test/iothub/device/MethodFaultInjectionTests.cs [hunk 6]
```diff
@@ -239,7 +239,7 @@ private async Task ServiceSendMethodAndVerifyResponseAsync(string deviceName, st
                     VerboseTestLogger.WriteLine($"{nameof(ServiceSendMethodAndVerifyResponseAsync)}: Method status: {response.Status}.");
 
                     response.Status.Should().Be(200);
-                    JsonSerializer.Serialize(response.Payload).Should().Be(JsonSerializer.Serialize(deviceResponsePayload));
+                    JsonSerializer.Serialize(response.PayloadAsString).Should().Be(JsonSerializer.Serialize(deviceResponsePayload));
```
Comment (abhipsaMisra) [inline e2e/test/iothub/device/MethodFaultInjectionTests.cs:242]:
```suggestion
                    response.PayloadAsString.Should().Be(JsonSerializer.Serialize(deviceResponsePayload));
```

**Pull request merged**